File: backend/VendorService/src/main/java/com/example/demo/entity/VendorSkillsTable.java
package com.example.demo.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "vendorskills_table")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class VendorSkillsTable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "venskill_id")
    private Integer venskillId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "vendor_id", nullable = false)
    private VendorTable vendor;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "skill_id", nullable = false)
    private SkillTable skill;
}

File: backend/admin_panel/Models/VendorskillsTable.cs
using System;
using System.Collections.Generic;

namespace admin_panel.Models;

public partial class VendorskillsTable
{
    public int VenskillId { get; set; }

    public int VendorId { get; set; }

    public int SkillId { get; set; }

    public virtual SkillTable Skill { get; set; } = null!;

    public virtual VendorTable Vendor { get; set; } = null!;
}
